### PR TITLE
Bv 61/refactor transactions list

### PR DESCRIPTION
**Pull request by Milena-Wasiak**
Connection with transactions data,
InfinityScroll
Style table

### CODE REVIEW

**Review by dariuszsobkowicz — APPROVED** [review]
well done 👍

File: src/components/TransactionsList/TransactionsList-container.tsx [hunk 1]
```diff
@@ -25,92 +21,43 @@ const headerCols: HeaderColsInterface[] = [
...
-    setPage(newPage);
-  };
+  let [pageNumber, setPageNumber]:[number, Function] = React.useState(1);
 
-  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
-    setRowsPerPage(parseInt(event.target.value, 10));
-    setPage(0);
-  };
 
   const handleClickOpen = (data: Record<string, any>) => {
```
Comment (dariuszsobkowicz) [inline src/components/TransactionsList/TransactionsList-container.tsx]: still in use?

File: src/components/BarChart/BarChart-container.tsx [hunk 2]
```diff
@@ -48,7 +48,12 @@ const BarChartContainer = (props: BarChartProps) => {
         <Link to={object.walletHash} key={object.walletHash}>
           <div
             className={getClasses(index)}
-            onClick={() => actions.fetchEthereumTransactionsSummed({walletHash:object.walletHash})}
+            onClick={
+              () => {
+                actions.flushEthereumTransactions();
+                actions.fetchEthereumTransactions({walletHash:object.walletHash, page: 1});
+                actions.fetchEthereumTransactionsSummed({walletHash:object.walletHash});
+            }}
```
Comment (damianlewandowski) [inline src/components/BarChart/BarChart-container.tsx]: This doesn't look like a good idea.

I wouldn't use onClick handlers here, completley unnecesary and is hard to think and reason about. This is an introduction to very messy codebase.

A much better solution is to read route params from the component where you need this data, and once it changes compare previous route param value to the current one, if it's different then run this action.

File: src/components/BarChart/BarChart-redux.tsx [hunk 3]
```diff
@@ -21,20 +22,35 @@ const BarChartRedux = () => {
...
             transactionsSummedData: transactionsSummedData
         });
     };
+    
+    const fetchEthereumTransactions = (transactionsData: TransactionsData): void => {
+        dispatch({
+            type: transactionsActions.ETHEREUM_FETCH_TRANSACTIONS,
+            transactionsData: transactionsData
+        });
+    };
```
Comment (damianlewandowski) [inline src/components/BarChart/BarChart-redux.tsx]: Why is this here? Ethereum transactions are not used in Barchart. Move it to TransactionsList component.

File: src/components/BarChart/BarChart-redux.tsx [hunk 4]
```diff
@@ -21,20 +22,35 @@ const BarChartRedux = () => {
...
           type: walletActions.ETHEREUM_FETCH_WALLETS
         });
       };
 
+    const flushEthereumTransactions = (): void => {
+        dispatch({
+          type: transactionsActions.ETHEREUM_FLUSH_TRANSACTIONS
+        });
+      };
+
```
Comment (damianlewandowski) [inline src/components/BarChart/BarChart-redux.tsx]: In what case do we need to clear the reducer state of transactions? Just replace the old value with the new one on route param change.

File: src/components/TransactionsList/TransactionsList-container.tsx [hunk 5]
```diff
@@ -1,22 +1,18 @@
...
 import Typography from '@material-ui/core/Typography';
 
-import TransactionListPagination from './TransactionsListPagination-view';
 import DetailsModal from './TransactionsListDetailsModal-view';
-import { stableSort, getSorting } from '../../helpers/helpers';
-import { HeaderColsInterface, ModalDetailsProps, Order, OrderBy } from './types';
+import { HeaderColsInterface, ModalDetailsProps, TransactionsListProps } from './types';
 import { timestampToDate } from './../../helpers/helpers';
-import { Block } from "../../types";
+import { transactionsListTableStyle } from './TransactionsList-styles';
```
Comment (damianlewandowski) [inline src/components/TransactionsList/TransactionsList-container.tsx]: This is a hook. A common convention for naming hooks is to use prefix `use` so name it `use TransactionsListTableStyle `

**Review by wojciech-nowakowski — APPROVED** [review]

File: src/components/TransactionsList/TransactionsList-container.tsx [hunk 6]
```diff
@@ -25,103 +20,58 @@ const headerCols: HeaderColsInterface[] = [
...
+      setPageNumber(pageNumber++);
+    }
+  }
+
+  const checkWalletHashAndFetchTransactions = (walletHash: any, pageNumber: number) => {
+    if (walletHash) {
+      actions.fetchEthereumTransactions({ walletHash: walletHash, page: pageNumber });
+    }
+  }
+
```
Comment (damianlewandowski) [inline src/components/TransactionsList/TransactionsList-container.tsx:63]: I know it may seem obvious what is happening here for now, but it won't in a couple of weeks / months.
Maybe some comment?

File: src/components/TransactionsList/TransactionsList-container.tsx [hunk 7]
```diff
@@ -25,103 +20,58 @@ const headerCols: HeaderColsInterface[] = [
...
+  useEffect((): void => {
+    checkWalletHashAndFetchTransactions(walletHash, pageNumber);
+  }, [pageNumber]);
+
```
Comment (damianlewandowski) [inline src/components/TransactionsList/TransactionsList-container.tsx:68]: Comment pls

**Review by damianlewandowski — APPROVED** [review]
https://i.kym-cdn.com/entries/icons/original/000/019/698/d96.jpg
